File: Data Structures & Algorithms/anagram-groups/submission-2.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap <String ,List<String>>map = new HashMap<>();
        if(strs == null || strs.length ==0)
        {
            return new ArrayList<>();
        }
        for (String word:strs){
            char [] temp = word.toCharArray();
            Arrays.sort(temp);
            String tep = new String (temp);
            if(!map.containsKey(tep)){
                map.put(tep, new ArrayList<>());
            }
            map.get(tep).add(word);
        }
        return new ArrayList<>(map.values());
    }
}

File: Data Structures & Algorithms/longest-common-prefix/submission-2.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        // string result="";
        string s=strs[0];
        for(int i=0; i<s.length();i++){
            for(const string &str:strs){
                if(s[i]!=str[i]){
                    return s.substr(0,i);
                }
            }
        }
        return s;
    }
};